README
Cairo::GObject
==============

Perl module to integrate Cairo into the Glib type system.

INSTALLATION
------------

To install this module type the following:

   perl Makefile.PL
   make
   make test
   make install

DEPENDENCIES
------------

This module requires these other modules and libraries:

  cairo-gobject (C library) >= 1.10.0
  Cairo (Perl module)       >= 1.080
  Glib (Perl module)        >= 1.224

In order to build it from source, you'll also need

  ExtUtils::Depends   >= 0.2
  ExtUtils::PkgConfig >= 1.0

BUG REPORTS
-----------

Please report bugs to the gnome-perl product in GNOME's Bugzilla:
<http://bugzilla.gnome.org/enter_bug.cgi?product=gnome-perl>

COPYRIGHT AND LICENSE
---------------------

Copyright (C) 2011 by the cairo perl team

Torsten Schoenfeld       kaffeetisch at gmx dot de

This library is free software; you can redistribute it and/or modify it under
the terms of the GNU Library General Public License as published by the Free
Software Foundation; either version 2.1 of the License, or (at your option) any
later version.

This library is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.  See the GNU Library General Public License for more
details.

You should have received a copy of the GNU Library General Public License along
with this library; if not, write to the Free Software Foundation, Inc., 59
Temple Place - Suite 330, Boston, MA  02111-1307  USA.


NEWS
Overview of changes in Cairo::GObject 1.003
===========================================

* Makefile.PL: add 'meta-spec' and 'license' keys to META_MERGE block to enable the expanded metadata display on MetaCPAN 

Overview of changes in Cairo::GObject 1.002
===========================================

* Makefile.PL: added CPAN::Meta::Spec v2 block to META_MERGE
* LICENSE: updated FSF address
* README: updated contact/mailing list/git info

Overview of changes in Cairo::GObject 1.001
===========================================

* Require Cairo 1.080 for the cairo_region_t converters.

Overview of changes in Cairo::GObject 1.000
===========================================

* Initial release.


lib/Cairo/GObject.pm
#
# Copyright (c) 2011 by the cairo perl team (see the file README)
#
# Licensed under the LGPL, see LICENSE file for more information.
#
# $Id$
#

package Cairo::GObject;

use strict;
use warnings;

use Cairo;
use Glib;

use XSLoader;

our $VERSION = '1.003';
XSLoader::load ('Cairo::GObject', $VERSION);

1;

__END__

=head1 NAME

Cairo::GObject - Integrate Cairo into the Glib type system

=head1 SYNOPSIS

  use Cairo::GObject;

  # Cairo and Glib are now loaded and the Cairo types are registed with
  # Glib's type machinery.  This allows you to correctly use Cairo types
  # in signals and properties.

=head1 ABSTRACT

Cairo::GObject registers Cairo's types (C<Cairo::Context>, C<Cairo::Surface>,
etc.) with Glib's type systems so that they can be used normally in signals and
properties.  If you have encountered an error akin to this:

  GType CairoContext (15497280) is not registered with gperl

-- then you need to use Cairo::GObject.

=head1 AUTHORS

=over

=item Torsten Schoenfeld E<lt>kaffeetisch at gmx dot deE<gt>

=back

=head1 COPYRIGHT

Copyright (C) 2011 by the cairo perl team

=cut


Makefile.PL
#
# Copyright (c) 2011 by the cairo perl team (see the file README)
#
# Licensed under the LGPL, see LICENSE file for more information.
#

BEGIN { require 5.008; }

use strict;
use warnings;
use ExtUtils::MakeMaker;

our %build_reqs = (
  'perl-ExtUtils-Depends'   => '0.2',
  'perl-ExtUtils-PkgConfig' => '1.0',
  'perl-Cairo'              => '1.080',
  'perl-Glib'               => '1.224',
  'cairo-gobject'           => '1.10.0',
);

our %pre_reqs = (
  'ExtUtils::Depends'   => $build_reqs{'perl-ExtUtils-Depends'},
  'ExtUtils::PkgConfig' => $build_reqs{'perl-ExtUtils-PkgConfig'},
  'Cairo'               => $build_reqs{'perl-Cairo'},
  'Glib'                => $build_reqs{'perl-Glib'},
);

our %meta_merge = (
    'meta-spec'          => {
        version => '2',
        url     => 'http://search.cpan.org/perldoc?CPAN::Meta::Spec',
    },
    'author' => ['gtk2-perl Team <gtk-perl-list at gnome dot org>'],
    # valid values: https://metacpan.org/module/CPAN::Meta::Spec#license
    license             => 'lgpl_2_1',
    'configure_requires'    => \%pre_reqs,
    'release_status'        => 'stable',
    'resources'   => {
        'bugtracker'    => {
            'mailto'        => 'bug-Cairo-GObject [at] rt.cpan.org',
            'web'           =>
            'https://rt.cpan.org/Public/Dist/Display.html?Name=Cairo-GObject',
        },
        'homepage'      => 'http://gtk2-perl.sourceforge.net/',
        'x_MailingList' => 'https://mail.gnome.org/mailman/listinfo/gtk-perl-list',
        'license'       => 'http://www.gnu.org/licenses/lgpl-2.1.html',
        'repository'    => {
            'type'          => 'git',
            'url'           => 'git://git.gnome.org/perl-Cairo-GObject',
            'web'           =>
                'http://git.gnome.org/browse/perl-Cairo-GObject/',
        },
    },
);

unless (eval "use ExtUtils::Depends '$build_reqs{'perl-ExtUtils-Depends'}';"
           . "use ExtUtils::PkgConfig '$build_reqs{'perl-ExtUtils-PkgConfig'}';"
           . "use Cairo '$build_reqs{'perl-Cairo'}';"
           . "use Glib '$build_reqs{'perl-Glib'}';"
           . "use Glib::MakeHelper;"
           . "1") {
  warn "$@\n";
  WriteMakefile(
    PREREQ_FATAL => 1,
    PREREQ_PM    => \%pre_reqs,
  );
  exit 1; # not reached
}

# If the package can't be found, warn and exit with status 0 to indicate to
# CPAN testers that their system is not supported.
my %pkgcfg;
unless (eval { %pkgcfg = ExtUtils::PkgConfig->find (
                 "cairo-gobject >= $build_reqs{'cairo-gobject'}");
	       1; })
{
	warn $@;
	exit 0;
}

mkdir 'build', 0777;

my $dep = ExtUtils::Depends->new('Cairo::GObject', qw/Cairo Glib/);
$dep->set_inc($pkgcfg{cflags});
$dep->set_libs($pkgcfg{libs});
$dep->add_xs('CairoGObject.xs');
$dep->add_pm('lib/Cairo/GObject.pm' => '$(INST_LIBDIR)/GObject.pm');
$dep->save_config('build/IFiles.pm');

WriteMakefile(
  NAME          => 'Cairo::GObject',
  VERSION_FROM  => 'lib/Cairo/GObject.pm',
  ABSTRACT_FROM => 'lib/Cairo/GObject.pm',
  PREREQ_PM     => \%pre_reqs,
  XSPROTOARG    => '-noprototypes',

  META_MERGE    => \%meta_merge,
  $dep->get_makefile_vars,
);

sub MY::postamble {
  return Glib::MakeHelper->postamble_clean ();
}
